SpiceSharp/Simulations/Exports/RealPropertyExport.cs

```diff
@@ -61,19 +56,19 @@ public RealPropertyExport(IEventfulSimulation simulation, string[] subcircuitEnt
         protected override void Initialize(object sender, EventArgs e)
         {
             e.ThrowIfNull(nameof(e));
-            if (SubcircuitEntityName == null)//then using entity name
+            if (EntityPath.Count == 1)//then path is entity name
             {
-                var eb = Simulation.EntityBehaviors[EntityName];
+                var eb = Simulation.EntityBehaviors[EntityPath[0]];
                 if (eb != null)
                     Extractor = eb.CreatePropertyGetter<double>(PropertyName);
             }
             else//using subcircuit array of strings
             {
-                string curComponentName = SubcircuitEntityName[0];
+                string curComponentName = EntityPath[0];
                 System.Collections.Generic.IEnumerator<Behaviors.IBehavior> entityEnum = Simulation.EntityBehaviors[curComponentName].GetEnumerator();
-                for (int i = 1; i < SubcircuitEntityName.Length; i++)
+                for (int i = 1; i < EntityPath.Count; i++)
                 {
-                    string nextComponentName = SubcircuitEntityName[i];
+                    string nextComponentName = EntityPath[i];
 
                     entityEnum.MoveNext();
```

Comment:
The issue here is that the algorithm will stop searching as soon as it find a behavior container for the subcircuit name, however, there can be multiple behavior containers for the same subcircuit name.
In this case, and in the unit tests, there is never a need to search further, so it works, but I don't think it will work in the most general case.

I propose something along the lines of this:

```csharp
// Collect all behaviors in the most top-level entity
var behaviors = new HashSet<IBehavior>();
foreach (var behavior in Simulation.EntityBehaviors[EntityPath[0]])
    behaviors.Add(behavior);

// For every subsequent name in the path, we need to collect the next set of behaviors
for (int i = 1; i < EntityPath.Count; i++)
{
    // Keep track of the behaviors one level deeper
    var subBehaviors = new HashSet<IBehavior>();

    // Go through all the behaviors of the previously found level, and collect the new set of behaviors
    foreach (var behavior in behaviors)
    {
        if (behavior is SpiceSharp.Components.Subcircuits.EntitiesBehavior subcktBehavior)
        {
            // Add all the behaviors in this one to the new level of found behaviors
            foreach (var subBehavior in subcktBehavior.LocalBehaviors)
                subBehaviors.Add(subBehavior);
        }        
    }

    // If we didn't find any new behaviors, then that means that the last level was not a subcircuit
    if (subBehaviors.Count == 0)
        throw new SpiceSharpException($"Entity {EntityPath[i - 1]} is not a subcircuit.");
    behaviors = subBehaviors; // Completed this level
}

// We have found all the behaviors for the relevant entity path, let's now find the property
// This code is basically identical to how it is implemented for BehaviorContainer
foreach (var behavior in behaviors)
{
    Extractor = behavior.CreatePropertyGetter<double>(PropertyName);
    if (Extractor != null)
        return; // Success
}

// If we reached this part, then none of the behaviors have defined the property...
throw new ParameterNotFoundException(this, PropertyName, typeof(double));
```

Reply:
Ok. integrated using your method. BTW, do you get notifications when I push a new commit?

Reply:
I get the notifications, but it's difficult to find the time to review at the moment.